### PR TITLE
Add optional group limit

### PR DESCRIPTION
**Pull request by colinleroy**
Hello,

I've been faced with a request at work where a domain had to be login-protected, and have used nginx-ldap-auth to be able to use our LDAP server.

Another request asked me to limit some of this domain's subdirectories to members of different departments.

This patch adds support for group limiting in nginx-ldap-auth. It uses a group fetch and checks the user's presence in the memberUid field of the LDAP group. I made this choice because some LDAP servers (including mine) badly support the memberOf overlay.

I hope it helps and you'll consider upstreaming that patch!

Colin

### CODE REVIEW

**Comment by vl-homutov** [comment]
If you still want to proceed with this, please provide tests.
The testsuite is here: https://github.com/nginxinc/nginx-ldap-auth/tree/master/t
Also, a good example for the doc would be nice with
description of which case for group auth is covered.

**Comment by colinleroy** [comment]
Hello, @vl-homutov 
I'm very sorry for the delay, but other priorities came up at work.
I've pushed one more commit to allow for multiple groups limit, and two more commits to address the documentation and the test suite.

I hope this helps !

File: README.md [hunk 1]
```diff
@@ -117,12 +117,27 @@ In addition, the **X-Ldap-Template** header can be used to create complex LDAP s
...
+```
+proxy_set_header X-Ldap-GroupBaseDN "ou=groups,dc=example,dc=com"
+proxy_set_header X-Ldap-GroupTemplate "(cn=%(groupname)s)"
+proxy_set_header X-Ldap-GroupLimit "group1"
+```
+
+The limit can be done on multiple groups using:
+
+```
+proxy_set_header X-Ldap-GroupLimit "group1,group2"
```
Comment (jmallach) [inline README.md:138]: Please add trailing semicolon to the nginx directive examples.

Comment (colinleroy) [reply README.md]: Hi,
Done :) Thanks!

**Comment by Derioss** [comment]
Hi, 
I have this script version on nginx server (more or less in production...)
And  I'm very interested in closing this MR.
What do you need for ?

**Pull request closed without merging**